### PR TITLE
Add renderAsync

### PR DESCRIPTION
**Pull request by RobLoach**
browserify-string.... Not sure I like it yet.

### CODE REVIEW

**Comment by TimothyGu** [comment]
Looks okay.

**Comment by RobLoach** [comment]
browserify-string adds an empty file to process.cwd().

**Comment by TimothyGu** [comment]
You should probably make that clear in the README, but still looks fine.

File: index.js [hunk 1]
```diff
@@ -20,3 +20,16 @@ exports.renderFileAsync = function (filename, options, locals) {
     });
   });
 };
+
+exports.renderAsync = function (str, options, locals) {
+  return new Promise(function (resolve, reject) {
+    var browserifyString = require('browserify-string');
```
Comment (RobLoach) [inline index.js:26]: @TimothyGu This is better.... It will only use `browserify-string` if it's needed.

**Pull request merged**